packages/jsii/lib/compiler.ts

```diff
@@ -98,10 +98,11 @@ export class Compiler implements Emitter {
         }
 
         const tsconf = this.typescriptConfig!;
+        const pi = this.options.projectInfo;
 
         const prog = ts.createProgram({
             rootNames: this.rootFiles.concat(_pathOfLibraries(this.compilerHost)),
-            options: COMPILER_OPTIONS,
+            options: {...COMPILER_OPTIONS, outDir: pi.tsc && pi.tsc.outDir, rootDir: pi.tsc && pi.tsc.rootDir},
```

Comment:
could be (since we decided to use the same name...):

```ts
{ ...pi.tsc, ...COMPILER_OPTIONS }
```

The order is important here - we don't want `pi.tsc` to be able to override the required settings we have in `COMPILER_OPTIONS`.

Reply:
I think for now this would work, but if someone adds rootDir or outDir to COMPILER_OPTIONS manual overrides would no longer be possible. But I am fine with the change.